mm2src/mm2_bitcoin/spv_validation/src/conf.rs

```diff
@@ -0,0 +1,132 @@
+use crate::helpers_validation::SPVError;
+use crate::work::{DifficultyAlgorithm, RETARGETING_INTERVAL};
+use chain::{BlockHeader, BlockHeaderBits};
+use primitives::hash::H256;
+use serde::Deserialize;
+use std::num::NonZeroU64;
+use std::str::FromStr;
+
+/// Custom SPV starting block header configuration
+#[derive(Debug, Clone, Deserialize)]
+pub struct StartingBlockHeader {
```

Comment:
`StartingBlockHeader` seems to be the same as `SPVVerificationHeader` but for the `bits` field. I think one struct for both is enough.

Reply:
`SPVVerificationHeader` is actually needed to minimize parsing of the `bit`s and `hash` to just one place.

for. example, `SPVVerificationHeader` is very helpful here, `StartingBlockHeader` isn't.
```rust
let mut previous_header: SPVVerificationHeader = if previous_height == conf.starting_block_header.height {
        conf.starting_block_header.to_verification_header(coin)?
    } else {
        storage
            .get_block_header(previous_height)
            .await?
            .ok_or(BlockHeaderStorageError::GetFromStorageError {
                coin: coin.to_string(),
                reason: format!("Header with height {} is not found in storage", previous_height),
            })?
            .into()
    };
```

Reply:
You can have only 
```rust
#[derive(Debug, Clone)]
pub struct SPVVerificationHeader {
    pub height: u64,
    pub hash: H256,
    pub time: u32,
    pub bits: BlockHeaderBits,
}
```
and implement deserialize for it